pages/[username]/conteudos/[page]/index.public.js

```diff
@@ -1,24 +1,41 @@
+import { useRouter } from 'next/router';
 import { getStaticPropsRevalidate } from 'next-swr';
 
-import { ContentList, DefaultLayout } from '@/TabNewsUI';
+import { ContentList, DefaultLayout, UserHeader } from '@/TabNewsUI';
+import { FaUser } from '@/TabNewsUI/icons';
 import { NotFoundError } from 'errors';
 import authorization from 'models/authorization.js';
 import content from 'models/content.js';
 import removeMarkdown from 'models/remove-markdown.js';
 import user from 'models/user.js';
 import validator from 'models/validator.js';
+import { useUser } from 'pages/interface';
 
 export default function Home({ contentListFound, pagination, username }) {
+  const { push } = useRouter();
+  const { user, isLoading } = useUser();
+  const isAuthenticatedUser = user && user.username === username;
+
   return (
-    <>
-      <DefaultLayout metadata={{ title: `Página ${pagination.currentPage} · ${username}` }}>
-        <ContentList
-          contentList={contentListFound}
-          pagination={pagination}
-          paginationBasePath={`/${username}/pagina`}
-        />
-      </DefaultLayout>
-    </>
+    <DefaultLayout metadata={{ title: `Publicações · Página ${pagination.currentPage} · ${username}` }}>
+      <UserHeader username={username} conteudosCount={pagination.totalRows} />
+
+      <ContentList
+        contentList={contentListFound}
+        pagination={pagination}
+        paginationBasePath={`/${username}/conteudos`}
+        emptyStateProps={{
+          isLoading: isLoading,
+          title: 'Nenhuma publicação encontrada',
+          description: `${isAuthenticatedUser ? 'Você' : username} ainda não fez nenhuma publicação.`,
```

Comment:
Pra mim, essa mensagem só faz sentido se estivermos na página 1.

Por exemplo, se eu navego para uma página alta, digamos a 1000, nenhum conteúdo será encontrado, mas não faz sentido dizer que eu ainda não fiz publicações, sendo que já fiz.

Talvez isso não precise ser mudado aqui, mas sim no `getStaticProps`. Acho que seria legal fazer um redirect para a última página válida sempre que a página for maior do que 1 e não for encontrado nenhum conteúdo.

Para esses casos, provavelmente deve funcionar se retornar algo como:

```js
  if (secureContentListFound.length === 0 && results.pagination.currentPage !== 1) {
    const lastValidPage = `/${secureUserFound.username}/conteudos/${results.pagination.lastPage || 1}`;
    const revalidate = results.pagination.currentPage > results.pagination.lastPage + 1 ? 60 : 1;

    return {
      redirect: {
        destination: lastValidPage,
      },
      revalidate,
    };
  }
```

Obs.: Não testei esse código e talvez compense refatorar. Talvez usar `context.params.page` no lugar de `results.pagination.currentPage`.